src/core/public/chrome/ui/header/recent_items.tsx

```diff
@@ -2,76 +2,129 @@
   EuiText,
+  EuiButtonEmpty,
   EuiSpacer,
   EuiHeaderSectionItemButtonProps,
   EuiButtonIcon,
+  EuiRadioGroup,
+  EuiTextColor,
+  EuiPopover,
   EuiToolTip,
 } from '@elastic/eui';
 import { i18n } from '@osd/i18n';
 import useObservable from 'react-use/lib/useObservable';
-import { ChromeRecentlyAccessedHistoryItem } from '../..';
+import { SavedObjectsNamespaceType } from 'src/core/public';
+import { ChromeRecentlyAccessedHistoryItem, SavedObject } from 'opensearch-dashboards/public';
 import { WorkspaceObject } from '../../../workspace';
-import { createRecentNavLink } from './nav_link';
 import { HttpStart } from '../../../http';
-import { ChromeNavLink } from '../../../';
 import './recent_items.scss';
 
+const widthForRightMargin = 8;
+
 export interface Props {
   recentlyAccessed$: Rx.Observable<ChromeRecentlyAccessedHistoryItem[]>;
   workspaceList$: Rx.Observable<WorkspaceObject[]>;
   navigateToUrl: (url: string) => Promise<void>;
-  basePath: HttpStart['basePath'];
-  navLinks$: Rx.Observable<ChromeNavLink[]>;
   renderBreadcrumbs: React.JSX.Element;
   buttonSize?: EuiHeaderSectionItemButtonProps['size'];
+  http: HttpStart;
 }
 
+interface SavedObjectMetadata {
+  icon?: string;
+  title?: string;
+  editUrl?: string;
+  inAppUrl?: { path: string; uiCapabilitiesPath: string };
+  namespaceType?: SavedObjectsNamespaceType;
+}
+export type SavedObjectWithMetadata<T = unknown> = SavedObject<T> & {
+  meta: SavedObjectMetadata;
+};
+
+type DetailedRecentlyAccessedItem = ChromeRecentlyAccessedHistoryItem &
+  SavedObjectWithMetadata &
+  ChromeRecentlyAccessedHistoryItem['meta'] & {
+    updatedAt: number;
+    workspaceName?: string;
+  };
+
+export const bulkGetDetail = (
+  savedObjects: Array<Pick<SavedObject, 'type' | 'id'>>,
+  http: HttpStart
+) => {
+  return Promise.all(
+    savedObjects.map((obj) =>
+      http
+        .get<SavedObjectWithMetadata>(
+          `/api/opensearch-dashboards/management/saved_objects/${encodeURIComponent(
+            obj.type
+          )}/${encodeURIComponent(obj.id)}`
+        )
+        .catch((error) => ({
+          id: obj.id,
+          type: obj.type,
+          error,
+          attributes: {},
+          references: [],
+          meta: {},
+          updated_at: '',
+        }))
+    )
+  );
+};
+
+const recentsRadios = [
+  {
+    id: 'recent_5',
+    label: '5 pages',
+  },
+  {
+    id: 'recent_10',
+    label: '10 pages',
+  },
+  {
+    id: 'recent_15',
+    label: '15 pages',
+  },
+];
+
 export const RecentItems = ({
   recentlyAccessed$,
   workspaceList$,
   navigateToUrl,
-  navLinks$,
-  basePath,
   renderBreadcrumbs,
   buttonSize = 's',
+  http,
 }: Props) => {
   const [isPopoverOpen, setIsPopoverOpen] = useState(false);
-
+  const [isPreferencesPopoverOpen, setIsPreferencesPopoverOpen] = useState(false);
   const recentlyAccessedItems = useObservable(recentlyAccessed$, []);
+  const [recentsRadioIdSelected, setRecentsRadioIdSelected] = useState('recent_5');
```

Comment:
What about `useState(recentsRadios[0].id) since recentsRadios is already a constant varible.

Reply:
Thank you, updated.